extensions/vertx/runtime/src/main/java/io/quarkus/vertx/VertxContextSupport.java

```diff
@@ -0,0 +1,98 @@
+package io.quarkus.vertx;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ManagedContext;
+import io.quarkus.vertx.core.runtime.VertxCoreRecorder;
+import io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle;
+import io.smallrye.common.vertx.VertxContext;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+
+/**
+ * Provides utility methods to work with Vertx duplicated context.
+ *
+ * @see VertxContext
+ */
+public final class VertxContextSupport {
+
+    private VertxContextSupport() {
+    }
+
+    /**
+     * Subscribes to the supplied {@link Uni} on a Vertx duplicated context; blocks the current thread and waits for the result.
+     * <p>
+     * If it's necessary, the CDI request context is activated during execution of the asynchronous code.
+     *
+     * @param uniSupplier
+     * @throws IllegalStateException If called on an event loop thread.
+     */
+    public static <T> T subscribeAndBlock(Supplier<Uni<T>> uniSupplier) throws Throwable {
```

Comment:
To be consistent with the Mutiny _await_ API, I would rename it to: `subscribeAndAwait()`

Reply:
OK